src/Generator/Passes/SymbolsCodeGenerator.cs

```diff
@@ -124,31 +180,27 @@ private void WrapConstructor(Method method)
             foreach (var param in method.Parameters.Where(
                 p => string.IsNullOrEmpty(p.OriginalName)))
                 param.Name = "_" + i++;
-            var @params = string.Join(", ", method.Parameters.Select(CastIfRVReference));
             var signature = string.Join(", ", method.GatherInternalParams(
                 Context.ParserOptions.IsItaniumLikeAbi).Select(
                     p => cppTypePrinter.VisitParameter(p)));
 
             string @namespace = method.Namespace.Visit(cppTypePrinter);
-            if (method.Access == AccessSpecifier.Protected)
+            Class @class = (Class) method.Namespace;
+            bool needSubclass = method.Access == AccessSpecifier.Protected || @class.IsAbstract;
+            if (needSubclass)
             {
-                Write(GetDerivedType(@namespace, wrapper));
-                Write(wrapper + @namespace);
-                Write($@"({string.Join(", ", method.Parameters.Select(
-                    p => cppTypePrinter.VisitParameter(p)))})");
-                WriteLine($": {@namespace}({@params}) {{}} }};");
                 Write($"extern \"C\" {{ void {wrapper}({signature}) ");
-                WriteLine($"{{ ::new ({Helpers.InstanceField}) {wrapper}{@namespace}({@params}); }} }}");
+                WriteLine($"{{ ::new ({Helpers.InstanceField}) {wrapper}{method.Namespace.Name}({@params}); }} }}");
             }
             else
             {
                 Write("extern \"C\" ");
-                if (method.Namespace.Access == AccessSpecifier.Protected)
+                if (needSubclass)
                     Write($@"{{ class {wrapper}{method.Namespace.Namespace.Name} : public {
                         method.Namespace.Namespace.Visit(cppTypePrinter)} ");
-                Write($"{{ void {wrapper}({signature}) ");
-                Write($"{{ ::new ({Helpers.InstanceField}) {@namespace}({@params}); }} }}");
-                if (method.Namespace.Access == AccessSpecifier.Protected)
+                Write($"{{ {GetExporting()}void {wrapper}({signature}) ");
+                Write($"{{ new ({Helpers.InstanceField}) {@namespace}({@params}); }} }}");
```

Comment:
Why did you remove the scope operator `::` from `new` here?

Reply:
No idea, I guess finding and replacing. Restored.